components/adblock_rust_ffi/src/lib.rs

```diff
@@ -373,3 +373,20 @@ pub unsafe extern "C" fn engine_hidden_class_id_selectors(
         .expect("Error: CString::new()")
         .into_raw()
 }
+
+#[cfg(feature = "ios")]
+#[no_mangle]
+pub unsafe extern "C" fn convert_rules_to_content_blocking(rules: *const c_char) -> *mut c_char {
+    let rules = CStr::from_ptr(rules).to_str().unwrap_or_else(|_| {
+        eprintln!("Failed to parse filter list with invalid UTF-8 content");
+        ""
+    });
+    let mut filter_set = adblock::lists::FilterSet::new(true);
+    filter_set.add_filter_list(&rules, Default::default());
+    // `unwrap` is safe here because `into_content_blocking` only panics if the `FilterSet` was not
+    // created in debug mode
+    let (cb_rules, _) = filter_set.into_content_blocking().unwrap();
+    CString::new(serde_json::to_string(&cb_rules).unwrap_or_else(|_| "".into()))
+        .expect("Error: CString::new()")
```

Comment:
Is this happening in a utility process, or will a panic here bring down the app? `CString::new` fails on interior null bytes; I certainly don't expect serde_json to produce that, but if there's a chance of crashing the app I'd rather see protection like
```rust
.unwrap_or_else(|| CString::new("").expect("CString::new() failed on an empty string))
```

Reply:
we've had that same pattern everywhere else in this FFI since it was created. Hasn't been causing issues so far, but I think it's best to let the migration to `cxx` handle all of that.